pkg/storage/ingest/pusher.go

```diff
@@ -339,6 +339,8 @@ func (p *shardingPusher) PushToStorage(ctx context.Context, request *mimirpb.Wri
 		s.Context = ctx // retain the last context in case we have to flush it when closing shardingPusher
 
 		if len(s.Timeseries) < p.batchSize {
+			// We modified a value so far. Write it back.
+			p.unfilledShards[shard] = s
```

Comment:
this is unrelated. I found the bug 2 weeks ago for some reason, so now I'm commiting it.

Reply:
I don' think we need this since the introduction of https://github.com/grafana/mimir/pull/9133 - I fixed it there.

Reply:
```suggestion
```